test/EngineTest.cpp

```diff
@@ -23,114 +32,259 @@ auto I = [](const auto& id) {
+      rightJC(RightJC),
+      testForSorted(TestForSorted),
+      sampleSolution(SampleSolution.clone())
+  {}
+};
 
-  for (size_t i = 1; i <= 10000; ++i) {
-    a.push_back({I(4 + i), I(2 + i)});
+/*
+ * @brief Tests, if the given IdTable has the same content as the sample
+ * solution and, if the option was choosen, if the IdTable is sorted by
+ * the join column.
+ *
+ * @tparam TABLE_WIDTH Just ignore it. Made to be infered from the function
+ *  paramters.
+ *
+ * @param table The IdTable that should be tested.
+ * @param sampleSolution The sample solution. Doesn't need to be sorted,
+ *  or the same order of rows as the table.
+ * @param testForSorted If this is true, it will also be tested, if the table
+ *  is sorted by the join column.
+ * @param jc The join column of the table.
+ * @param t Ignore it. It's only here for being able to make better messages,
+ *  if a IdTable fails the comparison.
+*/
+void compareIdTableWithSolution(const IdTable& table, 
+    const IdTable& sampleSolution,
+    const bool testForSorted = false,
+    const size_t jc = 0,
+    ad_utility::source_location t = ad_utility::source_location::current()
+    ) {
+  // For generating more informative messages, when failing the comparison.
+  std::stringstream traceMessage;
+  traceMessage << "compareIdTableWithSolution comparing IdTable\n";
+  for (size_t row = 0; row < table.size(); row++) {
+    for (size_t column = 0; column < table.numColumns(); column++) {
+      traceMessage << table(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  traceMessage << "with IdTable \n";
+  for (size_t row = 0; row < sampleSolution.size(); row++) {
+    for (size_t column = 0; column < sampleSolution.numColumns(); column++) {
+      traceMessage << sampleSolution(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  auto trace = generateLocationTrace(t, traceMessage.str());
+ 
+  // Because we compare tables later by sorting them, so that every table has
+  // one definit form, we need to create local copies.
+  IdTable localTable = table.clone();
+  IdTable localSampleSolution = sampleSolution.clone();
+
+  // Do the IdTable and sampleSolution have the same dimensions?
+  ASSERT_EQ(localTable.size(), localSampleSolution.size());
+  ASSERT_EQ(localTable.numColumns(), localSampleSolution.numColumns());
+
+  if (testForSorted) {
+    // Is the table sorted by join column?
+    auto oldEntry = localTable(0, jc);
+    for (size_t i = 1; i < localTable.size(); i++) {
+      ASSERT_TRUE(oldEntry <= localTable(i, jc));
+      oldEntry = localTable(i, jc);
+    }
+  }
+
+  // Sort both the table and the sampleSolution, so that both have a definite
+  // form for comparison.
+  auto sortFunction = [](
+      const auto& element1,
+      const auto& element2)
+    {
+      size_t i = 0;
+      while (i < (element1.numColumns() - 1) && element1[i] == element2[i]) {
+        i++;
+      }
+      return element1[i] < element2[i];
```

Comment:
The idea is good, but the rows of an IdTable are **not** iterable. I do not know, if it because of how data is stored, but the class for rows in IdTable has no function/support for creating iterators, or similiar to it, for the class, so you can't use any of the standard library functions, that you mentioned, with them.

They are simply not valide targets.

Reply:
The name change however, that one I did in [3ee5da2](https://github.com/ad-freiburg/qlever/pull/821/commits/3ee5da2715df951075eefde6e5a35c8edf9225ff).

Reply:
Added an TODO for this.